tests/test_objective_funs.py

```diff
@@ -485,25 +493,284 @@ def test_boundary_error_nestor(self):
+        zero_indices = np.where(eq.Rb_lmn == 0)
+        fixed_R_modes = eq.surface.R_basis.modes[zero_indices]
+
+        zero_indices = np.where(eq.Zb_lmn == 0)
+        fixed_Z_modes = eq.surface.Z_basis.modes[zero_indices]
+
+        constraints = (
+            ForceBalance(eq),
+            FixPressure(eq),
+            FixCurrent(eq),
+            FixPsi(eq),
+            FixBoundaryR(eq, modes=fixed_R_modes),
+            FixBoundaryZ(eq, modes=fixed_Z_modes),
+        )
+        quadflux_obj = QuadraticFlux(
+            ext_field=field,
+            eq=eq,
+            src_grid=source_grid,
+            eval_grid=eval_grid,
+            field_grid=field_grid,
+            field_fixed=True,
+        )
+        objective = ObjectiveFunction(quadflux_obj)
+
+        things_with_field_fixed, __ = nonlinear_optimizer.optimize(
+            eq,
+            objective=objective,
+            constraints=constraints,
+            gtol=0,
+            ftol=0,
+            copy=True,
+        )
+
+        new_Rb_lmn = things_with_field_fixed[0].Rb_lmn[
+            np.where(eq.surface.R_basis.modes[:, 2] != 0)
+        ]
+        new_Zb_lmn = things_with_field_fixed[0].Rb_lmn[
+            np.where(eq.surface.R_basis.modes[:, 2] != 0)
+        ]
+
+        np.testing.assert_allclose(new_Rb_lmn, 0, atol=1e-10)
+        np.testing.assert_allclose(new_Zb_lmn, 0, atol=1e-10)
+
+        # test with eq_fixed = field_fixed = False: nonaxisymmetric field and
+        # nonaxisymmetric equilibrium
+        eq = Equilibrium(M=2, L=2, N=4)
+        field = make_nonaxisym_field(eq)
+        field_grid, eval_grid, source_grid = get_grids(eq, field)
+        surface_grid = LinearGrid(
+            rho=np.array([1.0]),
+            M=20,
+            N=20,
+            NFP=int(eq.NFP),
+            sym=False,
+        )
+
+        quadflux_obj = QuadraticFlux(
+            ext_field=field,
+            eq=eq,
+            src_grid=source_grid,
+            eval_grid=eval_grid,
+            field_grid=field_grid,
+        )
+        plasma_vessel_obj = PlasmaVesselDistance(
+            surface=field,
+            eq=eq,
+            plasma_grid=source_grid,
+            surface_grid=surface_grid,
+            bounds=(0.2, 2),
+        )
+        mean_curvature_obj = MeanCurvature(eq, bounds=(-3, 3), grid=surface_grid)
+        objective = ObjectiveFunction(
+            (quadflux_obj, plasma_vessel_obj, mean_curvature_obj)
+        )
+
+        constraints = (
+            ForceBalance(eq),
+            FixPressure(eq),
+            FixCurrent(eq),
+            FixPsi(eq),
+            FixParameter(field, ["I", "G", "R_lmn", "Z_lmn"]),
+        )
+        things, __ = nonlinear_optimizer.optimize(
+            (field, eq),
+            objective=objective,
+            constraints=constraints,
+            ftol=0,
+            gtol=0,
+            copy=True,
+        )
+
+        np.testing.assert_allclose(things[0].Phi_mn, 0, atol=1e-7)
```

Comment:
This is basically just a repeat of test 1. I suggest evaluating $B \cdot n$ before and after the optimization, and checking that the errors decreased sufficiently. We don't want to force the solution to be axisymmetric for this test.

Reply:
It is not a repeat because the equilibrium is included in the optimization (as in, it is passed in and so this exercises the multiple object optimization framework)